test/integration/eds_integration_test.cc

```diff
@@ -25,6 +25,39 @@ class EdsIntegrationTest : public testing::TestWithParam<Network::Address::IpVer
       : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()),
         codec_client_type_(envoy::type::HTTP1) {}
 
+  // We need to supply the endpoints via EDS to provide health status. Use a
+  // filesystem delivery to simplify test mechanics.
+  void setEndpointsInPriorities(uint32_t first_priority, uint32_t second_priority,
+                                bool await_update = true) {
+    envoy::api::v2::ClusterLoadAssignment cluster_load_assignment;
+    cluster_load_assignment.set_cluster_name("cluster_0");
+
+    {
+      for (uint32_t i = 0; i < first_priority; ++i) {
+        auto* locality_lb_endpoints = cluster_load_assignment.add_endpoints();
+        locality_lb_endpoints->mutable_locality()->set_zone(&"1234"[i]);
```

Comment:
i think you might not need to set the zone at all? Does it still reproduce if we remove this line here and below?